serveradmin/nessus/views.py

```diff
@@ -0,0 +1,63 @@
+"""Serveradmin - Nessus Integration
+
+Copyright (c) 2023 InnoGames GmbH
+"""
+from urllib.parse import urlencode
+
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.http import (
+    HttpResponseBadRequest, HttpResponseServerError
+)
+from django.template.response import TemplateResponse
+from django.views.decorators.csrf import ensure_csrf_cookie
+
+from adminapi.filters import Any
+from serveradmin.dataset import Query
+from serveradmin.nessus.models import NessusAPI
+
+@login_required
+@ensure_csrf_cookie
+def nessus_config(request):
+    """Generate nessus scan configuration page
+
+    :param request:
+    :return:
+    """
+
+    hostnames = [h for h in request.GET.get('hosts', '').split(', ')]
+    object_ids = [o.strip() for o in request.GET.getlist('object_id', []) if o]
+
+    if len(hostnames) == 0 and len(object_ids) == 0:
+        return HttpResponseBadRequest('No hostname or object_id provided')
+
+    servers = {s['hostname']: s for s in
+               Query({'hostname': Any(*hostnames)}, None)}
+    servers.update({s['hostname']: s for s in
+                    Query({'object_id': Any(*object_ids)}, None)})
+
+    if request.GET.get('action') == 'Submit':
```

Comment:
You could think about making the form submit a POST request and use a form here to validate the submitted data.

See: https://docs.djangoproject.com/en/dev/ref/forms/api/

But since we don't do this yet everywhere and you use only options fields so values cannot be entered wrongly unless somebody does a HTTP request manually you can also leave it like this for now.